syntax = "proto3";

package flowpipe.v1.stages.kafka.source.v1;

// Configuration for KafkaSource stage
message KafkaSourceConfig {
  // Kafka bootstrap servers (comma-separated list)
  string brokers = 1;

  // Kafka topic name
  string topic = 2;

  // Consumer group id
  string group_id = 3;

  // Optional client.id
  string client_id = 4;

  // Optional partition (when set and >= 0, assign; otherwise subscribe)
  optional int32 partition = 5;

  // Optional auto.offset.reset (earliest/latest)
  string auto_offset_reset = 6;

  // Poll timeout in milliseconds
  int32 poll_timeout_ms = 7;

  // Additional librdkafka configuration options
  map<string, string> consumer_config = 8;
}
